platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/admin/ExportMigrationConfigurationAdminContext.java

```diff
@@ -0,0 +1,171 @@
+import org.codice.ddf.migration.ExportMigrationEntry;
+import org.codice.ddf.migration.MigrationEntry;
+import org.codice.ddf.migration.MigrationWarning;
+import org.codice.ddf.migration.ProxyExportMigrationContext;
+import org.osgi.service.cm.Configuration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ExportMigrationConfigurationAdminContext extends ProxyExportMigrationContext {
+    private static final Logger LOGGER = LoggerFactory.getLogger(
+            ExportMigrationConfigurationAdminContext.class);
+
+    private final ConfigurationAdminMigratable admin;
+
+    private final Set<String> warnedExtensions = new HashSet<>(8);
+
+    private final Map<Path, ExportMigrationEntry> entries;
+
+    public ExportMigrationConfigurationAdminContext(ExportMigrationContext context,
+            ConfigurationAdminMigratable admin, Configuration[] configs) {
+        super(context);
+        Validate.notNull(admin, "invalid null configuration admin migratable");
+        Validate.notNull(configs, "invalid null configurations");
+        this.admin = admin;
+        this.entries = Stream.of(configs)
+                .map(this::getEntry)
+                .collect(Collectors.toMap(MigrationEntry::getPath, Function.identity()));
+    }
+
+    public Stream<ExportMigrationEntry> entries() {
+        return entries.values()
+                .stream();
+    }
+
+    @Override
+    public ExportMigrationEntry getEntry(Path path) {
+        throw new IllegalStateException("should not be called");
+    }
+
+    @Override
+    public Stream<ExportMigrationEntry> entries(Path path) {
+        throw new IllegalStateException("should not be called");
+    }
+
+    @Override
+    public Stream<ExportMigrationEntry> entries(Path path, PathMatcher filter) {
+        throw new IllegalStateException("should not be called");
+    }
+
+    private ExportMigrationEntry getEntry(Configuration configuration) {
+        Validate.notNull(configuration, "invalid null configuration");
+        Path path = getPathFromConfiguration(configuration);
+        final String extn = FilenameUtils.getExtension(path.toString());
+        PersistenceStrategy ps = admin.getPersister(extn);
+
+        if (ps == null) {
+            ps = admin.getDefaultPersister();
+            if (warnedExtensions.add(extn)) {
+                getReport().record(new MigrationWarning(String.format(
+                        "Persistence strategy [%s] is not defined; defaulting to [%s]",
+                        extn,
+                        ps.getExtension())));
+            }
+            path = Paths.get(
+                    path.toString() + FilenameUtils.EXTENSION_SEPARATOR + ps.getExtension());
+        }
+        return new ExportMigrationConfigurationAdminEntry(super.getEntry(path), configuration, ps);
+    }
+
+    private Path getPathFromConfiguration(Configuration configuration) {
+        final Object o = configuration.getProperties()
```

Comment:
✏️  `filename` instead of just `o`?

Reply:
It ain't a filename until its type is assess later.